### PR TITLE
Problem: missing consequences in ADR-001 (fixes #492)

### LINKED ISSUE #492: Problem: missing consequences in ADR-001
https://github.com/crypto-org-chain/chain-main/blob/master/doc/architecture/adr-001.md

A few missing consequences:
- state sync not working in the current version / blocked by Cosmos SDK
- Windows and Arm64 compilation not working / blocked by Wasmer

I guess the decision may also need to be updated -- e.g. the CosmWasm module to be added under a feature flag until this is resolved + the future testnet version instructions may need to instruct users not to use state sync? @yihuang @leejw51crypto

### PR DESCRIPTION
**Pull request by tomtau**
Solution: updated the decision with more technical consequences
given the current state of CosmWasm.

### CODE REVIEW

File: doc/architecture/adr-001.md [hunk 1]
```diff
@@ -105,6 +110,8 @@ or operations on CosmWasm-originating digital assets.
 * The functionality in CosmWasm uploaded code may overlap with existing or future standardized native modules
 (i.e. potential extra complexity for wallets or other ecosystem tooling).
 * The attack surface is increased with the extra module.
+* "State Sync" may need to be disabled.
```
Comment (devashishdxt) [inline doc/architecture/adr-001.md:113]: Can "State Sync" be enabled in future versions of CosmWasm?

Comment (tomtau) [reply doc/architecture/adr-001.md:113]: it's blocked by Cosmos SDK https://github.com/cosmos/cosmos-sdk/issues/7340#issuecomment-811908201 at this moment

**Review by devashishdxt — APPROVED** [review]

File: doc/architecture/adr-001.md (hunk 1 above)
Comment (leejw51crypto) [reply doc/architecture/adr-001.md:113]: wasm code are stored on 
.chain-main/wasm/wasm/wasm/  
as files

cosmwasm statesync not possible, now

**Review by leejw51crypto — APPROVED** [review]
lgtm

**Pull request merged**